test/main.cpp

```diff
@@ -30,7 +34,51 @@
 
 using namespace std;
 
+static void set_filters_for_backend() {
+  std::string backend = "GENERIC";
+  char* pBE = getenv("NGRAPH_TF_BACKEND");
+  if (pBE) {
+    backend = std::string(pBE);
+    // backend.replace(backend.begin(), backend.end(), ":", "");
+    backend = std::regex_replace(backend, std::regex(":"), "");
+  }
+  std::string this_filepath = __FILE__;
+  this_filepath =
+      std::regex_replace(this_filepath, std::regex("^(.*)/.*$"), "$1");
+  std::string filter_file =
+      this_filepath + "/" + "cpp_tests_list_" + backend + ".txt";
+  fstream fs;
+  fs.open(filter_file, ios::in);
+  if (fs.is_open()) {
+    std::string filters = "";
+    std::string line;
+    bool excluded_section = false;
+    while (getline(fs, line)) {
+      line = std::regex_replace(line, std::regex("^\\s+"), "");
+      line = std::regex_replace(line, std::regex("#.*$"), "");
+      line = std::regex_replace(line, std::regex("\\s+$"), "");
```

Comment:
Probably not a big deal, but can combine the whitespace removal into one call using `remove_if`.

Reply:
First I strip off leading spaces, then all comments, then all remaining trailing spaces